src/datasources.js

```diff
@@ -95,13 +95,13 @@ define([
      *      have been loaded. This function should accept one argument,
      *      a list of data source objects.
      */
-    function getDataSources(callback) {
+    function getDataSources(successCallback, errorCallback) {
```

Comment:
Docstring should be updated. Also, `errorCallback` will be ignored on all calls to this function except the first one. Something like the `dataCallbacks` list seems necessary to fix that.